rpmlb/downloader/base.py

```diff
@@ -58,3 +68,25 @@ def after(self, work, **kwargs):
 
     def download(self, package_dict, **kwargs):
         raise NotImplementedError('Implement this method.')
+
+    def _is_download_skipped(self, package_dict: dict, **kwargs):
+        """Return if skip a download for a pacakge.
+
+        Override if needed.
+
+        Keyword arguments:
+            package_dict: A dictionary of package metadata.
+            kwargs: option arguments.
+        """
+
+        if not package_dict:
+            raise ValueError('package_dict is required.')
+
+        is_skipped = False
+
+        dist = kwargs.get('dist')
+        if dist and 'dist' in package_dict:
+            pattern = re.compile(package_dict['dist'])
+            if not pattern.match(dist):
```

Comment:
FYI: You do not need to compile the pattern for one-time use. This works just as well (and it it less clutter):

```python
if not re.match(package_dict['dist'], dist):
    ...
```

From [documentation](https://docs.python.org/3/library/re.html?#re.compile):

> The sequence
>
>     prog = re.compile(pattern)
>     result = prog.match(string)
>
> is equivalent to
> 
>     result = re.match(pattern, string)
>
> but using `re.compile()` and saving the resulting regular expression object for reuse is more efficient when the expression will be used several times in a single program.